deployer/src/deployer/upload.py

```diff
@@ -161,8 +161,15 @@ def read_chunks():
         return f'"{digests_md5.hexdigest()}-{len(md5s)}"'
 
     @property
-    def mime_type(self):
-        return mimetypes.guess_type(str(self.file_path))[0] or "binary/octet-stream"
+    def content_type(self):
+        mime_type = (
+            mimetypes.guess_type(str(self.file_path))[0] or "binary/octet-stream"
+        )
+        if mime_type.startswith("text/") or (
+            mime_type in ("application/json", "application/javascript")
+        ):
+            return f"{mime_type}; charset=utf-8"
```

Comment:
```suggestion
            mime_type += "; charset=utf-8"
```
because it's nicer to have fewer `return` statements unless they're used for early exits.